google/cloud/datastore/helpers.py

```diff
@@ -124,21 +108,18 @@ def entity_from_protobuf(pb):
     :rtype: :class:`google.cloud.datastore.entity.Entity`
     :returns: The entity derived from the protobuf.
     """
-    if not isinstance(pb, entity_pb2.Entity):
-        proto_pb = entity_pb2.Entity.wrap(pb)
```

Comment:
Is the removal of this line and focusing entirely on raw protobuf objects what delivers the final performance gains?

Reply:
Yes, indeed.  The proto-plus wrappers are an enormous perf-suck.  Most of the win in this PR is avoiding touching them anywhere except at the outermost entry point, and then unwrapping there ASAP.

Reply:
I had not pieced together that we could avoid `.wrap()` altogether. The last time I was in here, I'd mistakenly assumed calling that (which is at least the least-slow proto-plus entrypoint) was still required.

Very nice!